File: ui_automation_demoqa/src/main/java/pages/AlertsPage.java
package pages;

import com.microsoft.playwright.Page;

public class AlertsPage {
    private Page page;
    private String alertText;

    public AlertsPage(Page page) {
        this.page = page;
    }

    public void clickAlertButton() {
        // Setup alert handler to capture alert text
        page.onDialog(dialog -> {
            alertText = dialog.message();
            dialog.accept();
        });
        page.locator("#alertButton").click();
    }

    public String getAlertText() {
        return alertText;
    }

    public void acceptAlert() {
        // Alert is already accepted in the dialog handler
        // This method is kept for compatibility with the test
    }

    public void dismissAlert() {
        page.onDialog(dialog -> dialog.dismiss());
    }
}

File: ui_automation_demoqa_python/src/pages/alerts_page.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class AlertsPage:
    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 20)
        
        # Locators
        self.alert_button = (By.ID, "alertButton")
    
    def click_alert_button(self):
        """Click alert button"""
        element = self.wait.until(EC.element_to_be_clickable(self.alert_button))
        element.click()
    
    def get_alert_text(self):
        """Get alert text"""
        alert = self.wait.until(EC.alert_is_present())
        return alert.text
    
    def accept_alert(self):
        """Accept alert"""
        alert = self.wait.until(EC.alert_is_present())
        alert.accept()
    
    def dismiss_alert(self):
        """Dismiss alert"""
        alert = self.wait.until(EC.alert_is_present())
        alert.dismiss()